recipes/lora_finetune_distributed.py

```diff
@@ -468,6 +472,17 @@ def save_checkpoint(
                 intermediate_checkpoint=intermediate_checkpoint,
             )
 
+    def _should_update_weights(self, current_iteration: int) -> bool:
```

Comment:
This should be a utility at this point since it's in every recipe

Reply:
Yeah I just don't know where to put it tbh. None of the existing files feel relevant and I don't wanna create a new file just for this. Lmk if you have suggestions here

Reply:
maybe `torchtune/utils/training.py` or `torchtune/utils/optim_utils.py`? I don't see any harm in making a new file

Reply:
Tbh I'm inclined to go the other way and say that we should just do this inline. I definitely do not want to add even more indirection than we already have. And this is literally just doing a mod check

Reply:
Also @joecummings @kartikayk or @rohan-varma if you have thoughts here

Reply:
Do it inline.

Reply:
hm yeah, that's cleaner than I thought it would be. excellent choice